chain-impl-mockchain/src/transaction/witness.rs

```diff
@@ -56,8 +56,13 @@ impl std::fmt::Display for Witness {
 pub struct WitnessUtxoData(Vec<u8>);
 
 impl WitnessUtxoData {
-    pub fn new(block0: &HeaderId, transaction_id: &TransactionSignDataHash) -> Self {
+    pub fn new(block0: &HeaderId, transaction_id: &TransactionSignDataHash, is_legacy: bool) -> Self {
```

Comment:
this shouldn't be a bool, something like (not keen on "Normal" but couldn't think of anything else):

```pub enum UtxoWitnessType { Normal, Legacy } ```